src/pages/Inventory.page.ts

```diff
@@ -0,0 +1,36 @@
+// @ts-check
+import BaseSwagLabPage from './BaseSwagLab.page';
+
+class InventoryPage extends BaseSwagLabPage {
+    url = '/inventory.html';
+
+    headerTitle = this.page.locator('.title');
+
+    inventoryItems = this.page.locator('.inventory_item');
+
+    addItemToCartButton = this.page.locator('[id^="add-to-cart"]');
+
+    sortDropdown = this.page.locator('select.product_sort_container');
+
+    async addItemToCartById(id) {
+        await this.addItemToCartButton.nth(id).click();
+    }
+
+    async sortBy(optionText) {
+        await this.sortDropdown.selectOption({ value: optionText });
+    }
+
+    async getAllProducts() {
+        return this.inventoryItems.evaluateAll((items) => items.map((item) => ({
```

Comment:
I would use standard PW methods, something like this:

async getAllProducts() {
        const allProducts = await this.inventoryItems.all();
        const randomProducts = getRandomLocators(allProducts);

        const products = [];

        for await (const element of randomProducts) {
            const name = await element.getByTestId('inventory-item-name').innerText();
            const description = await element
                .getByTestId('inventory-item-desc')
                .innerText();
            const price = await element
                .getByTestId('inventory-item-price')
                .innerText();

            products.push({ name, description, price });
        }

        return products;
}